bin/ocaml/ocaml_merlin.ml

```diff
@@ -129,48 +133,61 @@ end = struct
       |> error
   ;;
 
-  let to_local file =
+  let to_local ~selected_context file =
     match to_local file with
     | Error s -> Fiber.return (Error s)
     | Ok file ->
-      let+ workspace = Memo.run (Workspace.workspace ()) in
       let module Context_name = Dune_engine.Context_name in
-      (match workspace.merlin_context with
-       | None -> Error "no merlin context configured"
-       | Some context ->
-         Ok (Path.Build.append_local (Context_name.build_dir context) file))
+      (match selected_context with
+       | Custom ctxt ->
+         Fiber.return (Ok (Path.Build.append_local (Context_name.build_dir ctxt) file))
+       | Default ->
+         let+ workspace = Memo.run (Workspace.workspace ()) in
+         (match workspace.merlin_context with
+          | None -> Error "no merlin context configured"
+          | Some context ->
+            Ok (Path.Build.append_local (Context_name.build_dir context) file)))
   ;;
 
-  let print_merlin_conf file =
-    to_local file
+  let print_merlin_conf ~selected_context file =
+    to_local ~selected_context file
     >>| (function
            | Error s -> Merlin_conf.make_error s
            | Ok file -> load_merlin_file file)
     >>| Merlin_conf.to_stdout
   ;;
 
   let dump s =
-    to_local s
+    to_local ~selected_context:Default s
     >>| function
     | Error mess -> Printf.eprintf "%s\n%!" mess
     | Ok path -> get_merlin_files_paths path |> List.iter ~f:Merlin.Processed.print_file
   ;;
 
   let dump_dot_merlin s =
-    to_local s
+    to_local ~selected_context:Default s
     >>| function
     | Error mess -> Printf.eprintf "%s\n%!" mess
     | Ok path ->
       let files = get_merlin_files_paths path in
       Merlin.Processed.print_generic_dot_merlin files
   ;;
 
-  let start () =
+  let start ~ctx_name () =
+    let open Fiber.O in
+    let selected_context =
+      match ctx_name with
+      | None -> Default
+      | Some ctx_name ->
+        (match Context_name.of_string_opt ctx_name with
```

Comment:
I was wondering what the valid names for contexts were and incidentally found a crash: https://github.com/ocaml/dune/pull/10472